### PR TITLE
Issue 51

### PR DESCRIPTION
**Pull request by jice73**
Documentation updates and fixes for #51 #8

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,42 +1,45 @@
-cmake_minimum_required(VERSION 3.0...3.16)
-project(OPENNS LANGUAGES C CXX VERSION 2.0.0)
+cmake_minimum_required(VERSION 3.15)
```
Comment (giumas) [inline CMakeLists.txt:1]: Why are you forcing 3.15 as minimum CMake version? What do we specifically need from 3.15?

Comment (jice73) [reply CMakeLists.txt:1]: Here's the things we are using from various CMake versions.

3.1
    CXX_STANDARD

3.7
    "Visual Studio 15 2017" generator

3.8
    UseSWIG module gained swig_add_library()

3.12
    new FindPython3 module
    UseSWIG updates

3.13
    added cmake -S .
    added cmake -B build

3.15
    CMP0092 (remove /W3 from default MSVC compiler flags)

Comment (giumas) [reply CMakeLists.txt:1]: I have concerns about this approach to pushing the minimum CMake to a version released only last July. Specially, it is for a minor feature: https://gitlab.kitware.com/cmake/cmake/commit/1baf122cd41d6500139649661052236b8e684565

File: .travis.yml [hunk 2]
```diff
@@ -42,5 +42,8 @@ script:
   # C++ tests.
   - build/tests/bag_tests --samples-path examples/sample-data
   # Python tests.
-  - LD_LIBRARY_PATH=/home/travis/build/jice73/BAG/build/api PYTHONPATH=/home/travis/build/jice73/BAG/build/api/swig_i python python/test_all.py
+  # HOME == /home/travis
+  # ? == build/jice73 (some kind of git/checkout/github variable?)
+  - env
+  #- LD_LIBRARY_PATH=/home/travis/build/jice73/BAG/build/api PYTHONPATH=/home/travis/build/jice73/BAG/build/api/swig_i python python/test_all.py
```
Comment (giumas) [inline .travis.yml]: Doesn't having 'jice73' makes it only working with your Travis account?

Comment (jice73) [reply .travis.yml]: It does.  There is no mention of jice73 in the latest version of the.travis.yml file.  :)

**Comment by jice73** [comment]
> I have concerns about adopting a relatively recent CMake version (July 2019) just for a minor feature: https://gitlab.kitware.com/cmake/cmake/commit/1baf122cd41d6500139649661052236b8e684565


Sure.  What version should have I targette?

File: .travis.yml (hunk 2 above)
Comment (giumas) [reply .travis.yml]: I am confused by the commented rows! What is the point of keeping them?

Comment (jice73) [reply .travis.yml]: They aren't kept; that's an old version. of the file.

File: examples/CMakeLists.txt [hunk 3]
```diff
@@ -1,5 +1,5 @@
-cmake_minimum_required(VERSION 3.0...3.16)
-project(OpenNS_Examples LANGUAGES C CXX)
+cmake_minimum_required(VERSION 3.15)
```
Comment (giumas) [inline examples/CMakeLists.txt:1]: why do we need multiple instance of 'cmake_minimum_required'? isn't enough to declare it in the root's CMakeLists?

File: .travis.yml [hunk 4]
```diff
@@ -36,11 +36,11 @@ install:
...
 script:
-  - cmake -G Ninja -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=$HOME/install -DBUILD_PYTHON:BOOL=ON -B build -S .
+  - cmake -G Ninja -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=$HOME/install -DBAG_BUILD_PYTHON:BOOL=ON -B build -S .
   - cmake --build build
   - cmake --install build
   # C++ tests.
   - build/tests/bag_tests --samples-path examples/sample-data
   # Python tests.
-  - LD_LIBRARY_PATH=/home/travis/build/jice73/BAG/build/api PYTHONPATH=/home/travis/build/jice73/BAG/build/api/swig_i python python/test_all.py
+  - LD_LIBRARY_PATH=$TRAVIS_BUILD_DIR/build/api PYTHONPATH=$TRAVIS_BUILD_DIR/BAG/build/api/swig_i python python/test_all.py
```
Comment (giumas) [inline .travis.yml:45]: I am confused by this required step. In Python you don't usually modify paths before running a script.

File: docs/sub/linux.html [hunk 5]
```diff
@@ -1,5 +1,23 @@
...
+    <br>
+    <span style="color: red;">cmake -G "Unix Makefiles" -B build -S . -DCMAKE_BUILD_CONFIG=Release -DCMAKE_INSTALL_PREFIX=/usr/local</span><br>
+    <br>
+    To add support for Python, SWIG 4.0.1 or newer is needed.  Add <span style="color: red;">-DBAG_BUILD_PYTHON=ON</span> to the above command.<br>
+    <br>
+    The makefile is now available in the bag/build folder. To build, simply type:<br>
+    <br>
+    <span style="color: red;">cmake --build build</span><br>
+    <br>
+    To install, simply type:<br>
```
Comment (giumas) [inline docs/sub/linux.html:19]: This is the third 'simply' in just a few rows. Should we remove the last two ones?

File: docs/sub/mingw.html [hunk 6]
```diff
@@ -1,11 +0,0 @@
-<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">
```
Comment (giumas) [inline docs/sub/mingw.html:1]: Why did we drop MingWin support?

File: docs/sub/visual_studio_2017.html [hunk 7]
```diff
@@ -0,0 +1,72 @@
+<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">
+<html>
+<head>
+	<meta content="text/html; charset=ISO-8859-1" http-equiv="content-type">
+	<title>linux</title>
+</head>
+<body>
+	<h1>Building with Visual Studio 2017</h1>
+	Note:  It is assumed you install your third party libraries into C:\third_party.<br>
```
Comment (giumas) [inline docs/sub/visual_studio_2017.html:9]: Why do we have instructions of how to install the dependencies on Windows? If there are not special instructions to follow, we should not have those details. For instance, I am using Anaconda on Windows and I don't need to build the dependencies following these instructions. Incidentally, it is the same approach followed in the AppVeyor CI.

File: readme.rst [hunk 8]
```diff
@@ -12,7 +12,25 @@ work with data in the BAG format:
...
+Doxygen is needed to produce the documentation.  Get it at http://www.doxygen.nl/
+
+::
+    cd docs/api
+    doxygen docs_config.dox
+
+  Browse to html/index.html
+Running C++ tests in Visual Studio
+----------------------------------
+The C++ unit tests are written with Catch2.  There is no test harness at the moment.  Using the Standard toolbar, choose bag_tests.exe as the Select Startup Item.  Press F5 to run the tests.
```
Comment (giumas) [inline readme.rst:32]: Do you mean the 'Solution Explorer'? I don't understand how to use the 'Standard toolbar' for this task.
Did you forget the instructions about setting the '--samples-path'?

File: readme.rst [hunk 9]
```diff
@@ -12,7 +12,25 @@ work with data in the BAG format:
...
+
+Running Python tests in Visual Studio
+-------------------------------------
+The Python tests do not use any unit test framework at the moment.  Using the Standard toolbar, choose Current Document as the Select Startup Item.  Open test_all.py (or any other test_*.py) in the python folder, and press F5 to run the tests.
```
Comment (giumas) [inline readme.rst:36]: Same comment as the previous one.

**Pull request merged**